poetry/utils/env.py

```diff
@@ -1117,11 +1117,13 @@ def site_packages(self) -> SitePackages:
     def usersite(self) -> Optional[Path]:
         if "usersite" in self.paths:
             return Path(self.paths["usersite"])
+        return None
```

Comment:
@abn 
explicitly returning `None` is required to conform with PEP8 and to satisfy mypy

from PEP8-

> Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should. If any return statement returns an expression, any return statements where no value is returned should explicitly state this as return None, and an explicit return statement should be present at the end of the function (if reachable)

it's not without controversy, see this discussion - https://github.com/python/mypy/issues/7511

it depends what the goal is. The goal of this pull request is not strictly correctness, it is to reduce the amount of noise that mypy makes, in the hope that this is a small step in the direction of actually being able to effectively use Mypy to lint this codebase (and *also* correctness!).

In a perfect world, the number of type errors would be 0, and then mypy could be added as a pre-commit hook or CI job to ensure type errors are not reintroduced. Baby steps.